### PR TITLE
chore: update `ic-cdk` to 0.16.0

### PR DESCRIPTION
**Pull request by gregorydemay**
Relevant breaking changes from [`ic-cdk`](https://github.com/dfinity/cdk-rs/blob/main/src/ic-cdk/CHANGELOG.md):
1. [[0.15.0]]([0.15.0]): 
    1. BREAKING: Stable Memory always use 64-bit addresses and stable64_* system API. (dfinity/cdk-rs#498)
    2. BREAKING: Add log_visibility to the management canister API types: (dfinity/cdk-rs#497)
        - CanisterSettings
        - DefiniteCanisterSettings.
3. [[0.13.3]](https://github.com/dfinity/cdk-rs/blob/main/src/ic-cdk/CHANGELOG.md#0133---2024-05-10-yanked) BREAKING: Add wasm_memory_limit to the management canister API types: (dfinity/cdk-rs#483)
    - CanisterSettings
    - DefiniteCanisterSettings.

### CODE REVIEW

**Review by berestovskyy — APPROVED** [review]
LGTM, thanks!

**Review by mbjorkqvist — APPROVED** [review]
Thanks @gregorydemay !

**Review by mraszyk — APPROVED** [review]

**Review by ninegua — APPROVED** [review]

**Review by rumenov — APPROVED** [review]
Thanks for doing this!

**Review by basvandijk — APPROVED** [review]

File: rs/nns/sns-wasm/src/canister_stable_memory.rs [hunk 1]
```diff
@@ -9,37 +8,21 @@ use std::sync::{Arc, Mutex};
 pub struct CanisterStableMemory {}
 
 impl StableMemory for CanisterStableMemory {
```
[the remark this answers is not in the document]
Comment (daniel-wong-dfinity-org) [reply rs/nns/sns-wasm/src/canister_stable_memory.rs:10]: I think this is for dependency injection.

**Review by daniel-wong-dfinity-org — APPROVED** [review]
Approving for NNS.

File: rs/nns/sns-wasm/src/canister_stable_memory.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (gregorydemay) [reply rs/nns/sns-wasm/src/canister_stable_memory.rs:10]: I think you could replace this struct `CanisterStableMemory` and its usages with `ic_cdk::api::stable::CanisterStableMemory` without any change of behaviour (didn't actually test it so maybe I'm missing something) but I will leave that to the NNS team.

**Pull request merged**